### PR TITLE
MES-3944 - Schema validation

### PR DESCRIPTION
**Pull request by matthew2564**
# Description and relevant Jira numbers

## Pull Request checklist

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop

## Sign off process checklist

- [ ] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [ ] Reviewers selected in Github
- [x] PR link added to JIRA

### CODE REVIEW

File: src/functions/postResult/domain/mes-joi-schema-service.ts [hunk 1]
```diff
@@ -1,16 +1,46 @@
...
+ * @param {TestResultSchemasUnion} testResult
+ * @return {string}
+ */
+export const getTestCategory = (testResult: TestResultSchemasUnion): string => testResult.category;
+
+/**
+ * Function to return the desired schema to validate against
+ * @param {string} category
+ */
+export const getCategorySpecificSchema = (category: string) => {
```
Comment (ammarhaiderbjss) [inline src/functions/postResult/domain/mes-joi-schema-service.ts]: return type here?

File: serverless.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-service: mes-microservices-starter-template
+service: mes-result-service
```
Comment (ammarhaiderbjss) [inline serverless.yml:1]: nice spot!

File: src/functions/postResult/domain/mes-joi-schema-service.ts (hunk 1 above)
Comment (ammarhaiderbjss) [reply src/functions/postResult/domain/mes-joi-schema-service.ts]: also might have been nice to put this in its own file

**Review by ammarhaiderbjss — APPROVED** [review]

**Review by rossfauldsbjss — APPROVED** [review]
looks good.

**Review by rstepnowski — APPROVED** [review]

**Pull request merged**